File: server/app/api/routes/focus.py
from __future__ import annotations

from fastapi import APIRouter, Depends, Query

from app.api.routes.auth import get_current_user
from app.features.focus.schemas import FocusMutationResponse, FocusOverviewResponse, FocusWatchRequest
from app.features.focus.service import service

router = APIRouter()


@router.get("/overview", response_model=FocusOverviewResponse)
async def get_focus_overview(
    job_id: str | None = Query(default=None),
    current_user=Depends(get_current_user),
):
    return service.get_overview(current_user=current_user, selected_job_id=job_id)


@router.post("/watch", response_model=FocusMutationResponse)
async def watch_account(payload: FocusWatchRequest, current_user=Depends(get_current_user)):
    return service.watch_account(current_user=current_user, payload=payload)


@router.delete("/watch/{account}", response_model=FocusMutationResponse)
async def unwatch_account(account: str, current_user=Depends(get_current_user)):
    return service.unwatch_account(current_user=current_user, account=account)


@router.delete("/logs/{job_id}", response_model=FocusMutationResponse)
async def hide_focus_log(job_id: str, current_user=Depends(get_current_user)):
    return service.hide_log(current_user=current_user, job_id=job_id)

File: desktop/src/api/modules/focus.ts
import { apiDeleteJson, apiGet, apiPost } from "@/api/client";
import type { FocusMode, FocusMutationResponse, FocusOverviewResponse } from "@/types/focus";

export function fetchFocusOverview(jobId?: string) {
  const query = jobId ? `?job_id=${encodeURIComponent(jobId)}` : "";
  return apiGet<FocusOverviewResponse>(`/api/focus/overview${query}`);
}

export function addFocusAccount(payload: { account: string; mode: FocusMode; job_id?: string | null }) {
  return apiPost<FocusMutationResponse>("/api/focus/watch", payload);
}

export function removeFocusAccount(account: string) {
  return apiDeleteJson<FocusMutationResponse>(`/api/focus/watch/${encodeURIComponent(account)}`);
}

export function hideFocusLog(jobId: string) {
  return apiDeleteJson<FocusMutationResponse>(`/api/focus/logs/${encodeURIComponent(jobId)}`);
}
